Add exporters for Harmor and Sytrus

Azorlogh opened the pull request:
This PR adds exporters for Image-Line's Harmor and Sytrus, as well as docs for how to use them.
While Harmor does support .scl import, it doesn't allow changing the base frequency or MIDI note, making it hard to use in some projects.
Sytrus on the other hand does not have .scl import at all.

However both synths have a "pitch keyboard mapping" envelope that can offset pitches for each MIDI key from C0 to C10.
These exporters add support for .fnv files, which are the mapping files for these synths.

Harmor and Sytrus each need different .fnv files to work, because Harmor allows retuning up or down 5 octaves from 12 EDO, while sytrus is only 4 octaves up or down.

I have also added an optional `raw` boolean parameter to saveFile, to allow saving binary data (Uint8Array) instead of a string.

Let me know if any changes are needed.

meszaros-lajos-gyorgy commented on src/js/exporters.js:
```diff
@@ -272,6 +276,57 @@ function exportKontaktScript() {
+    const edo12cents = (i - 69) * 100
+    const offset = tuningTable.cents[i] - edo12cents
+    const normalizedOffset = ((offset / 1200 + baseFreqOffset) / range) * 0.5 + 0.5
+    const yCoord = Math.max(0.0, Math.min(1.0, normalizedOffset))
```
My eyes cought this line of code, which could be a simplified a bit using the [clamp](https://github.com/SeanArchibald/scale-workshop/blob/develop/src/js/helpers/numbers.js#L233) function from `helpers/numbers.js`:

```javascript
const yCoord = clamp(0, 1, normalizedOffset)
```

meszaros-lajos-gyorgy commented on src/js/exporters.js:
```diff
@@ -272,6 +276,57 @@ function exportKontaktScript() {
+    const yCoord = Math.max(0.0, Math.min(1.0, normalizedOffset))
+    pointsDoubles[i * 3 + 1] = yCoord
+    if (i !== 0) { // no x offset and no curve data for first point
+      pointsDoubles[i * 3] = 0.0083333337679505 // constant x offset from previous point
```
I suggest moving that fraction out to a constant similarly to `HEADER_BYTES` and `ENDING_BYTES`. It would decrease the "magic number" factor by giving context to it and would inform people where this number is coming from. Also, if this value can be calculated via some math, then I would definitely prefer that compared to having an arbitrary length number.

meszaros-lajos-gyorgy commented on src/js/exporters.js:
```diff
@@ -272,6 +276,57 @@ function exportKontaktScript() {
+    if (i !== 0) { // no x offset and no curve data for first point
+      pointsDoubles[i * 3] = 0.0083333337679505 // constant x offset from previous point
+      pointsUint32[i * 6 + 4] = 0        // |
+      pointsUint32[i * 6 + 5] = 33554432 // | no curve
```
This raises the same question as before: where is this number coming from and can this be calculated? Please also move this number out to a constant to give it some context.

meszaros-lajos-gyorgy reviewed (changes requested):
Thank you very much for adding these features! The code looks good to me, except for having those constant numbers baked into the code. Please moved those out to constants and then I'll merge your code.

SeanArchibald commented:
@Azorlogh thanks so much for doing this. I know a bunch of FL Studio users and they will find this feature useful!

Azorlogh commented:
Thanks for the feedback!
I have added constants and calculations where it's possible, and I've also reverted changes that my formatter did to some unrelated part of code for some reason.
Let me know if any further changes need to be made.

meszaros-lajos-gyorgy commented:
Merged your PR, thank you very much for the feature!
